sklearn/linear_model/_base.py

```diff
@@ -46,6 +46,30 @@
 # intercept oscillation.
 
 
+def _deprecate_normalize(normalize, default):
+    if normalize == 'deprecated':
+        _normalize = default
+    else:
+        _normalize = normalize
+
+    if default or (normalize != 'deprecated' and normalize):
```

Comment:
I don't think the `if default` part is correct:

if the default is True, we'll need 2 deprecation cycles I think: one fore changing the default and one for removing the parameter. Otherwise, there is no way for users to either have no warning or to not break their code.

Thouoghts @agramfort @glemaitre ?

Reply:
I have a doubt too now. @maikia can you write a tiny test for this private function?

Reply:
sure I will @agramfort 
But I think the question of @NicolasHug is not if it works, but if we should not change the default to False before deprecation of 'normalize' in the models when the default is True.